blog/2022-maintainer-summit.md

```diff
@@ -0,0 +1,121 @@
+
+This proposal outlines a new layer of Electron C APIs that will allow app developers to write their
+own Native Node Addons that interface with Electron’s internal resources, similar to Node’s
+own Node-API.
+
+#### Example: Supercharging apps with Chromium resources
+
+Many Electron apps maintain their own forks to interact directly with Chromium internals that would
+otherwise be inaccessible with vanilla (unmodified) Electron. By exposing these resources in the C
+API layer, this code can instead live as a native module alongside Electron, potentially reducing
+app developer maintenance burden.
+
+### Exposing Chromium’s UI layer (Views API)
+
+Under the hood, the non-website parts of Chrome’s user interface (UI), such as toolbars, tabs, or
+buttons, are built with a framework called Views. The Views API proposal introduces parts of this
+framework as JavaScript classes in Electron, with the eventual goal of allowing developers to create
+non-web UI elements to their Electron applications. This will prevent apps from having to hack
+together web contents.
+
+The groundwork to make this new set of APIs possible is currently in progress. Here are a few of the
+first things you can expect in the near future.
+
+#### Example: Refactoring the window model with `WebContentsView`
+
+Our first planned change is to expose Chrome’s WebContentsView to Electron’s API surface, which will
+be the successor to our existing BrowserView API (which, despite the name, is Electron-specific code
+unrelated to Chromium Views). With WebContentsView exposed, we will have a reusable View object that
+can display web contents, opening the door to making the BrowserWindow class pure JavaScript and
+eliminating even more code complexity.
+
+Although this change doesn’t provide a lot of new functionality to app developers, it is a large
+refactor that eliminates a lot of code under the hood, simplifying Chromium upgrades and reducing
+the risk of new bugs appearing between major versions.
+
+If you’re an Electron developer using BrowserViews in your app: don’t worry, we haven’t forgotten
+about you! We plan on making the existing BrowserView class a shim for WebContentsView to provide a
+buffer as you transition to the newer APIs.
+
+See: electron/electron#35658
+
+#### Example: Scrollable web contents with `ScrollView`
+
+Our friends at Stack have been driving an initiative to expose the Chromium ScrollView component to
+Electron’s API. With this new API, any child View component can be made scrollable horizontally or
+vertically.
+
+Although this new API fulfills a single smaller functionality, the team’s eventual goal is to build
+a set of utility View components that can be used as a toolkit to build more complex non-HTML
+interfaces.
+
+### Getting involved
+
+Are you an Electron app developer interested in either of these API proposals? Although we’re not
+quite ready to receive additional RFCs, please stay tuned for more details in the future!
```

Comment:
Do we have an official process for RFCs already? Was looking in governance, but that might be a cool thing to link to/describe here.

Reply:
Our current one isn't tailored for external RFCs. I wanted to propose something to the larger governance crowd but haven't gotten to it yet.